04-cs-fundamentals/classroom/Binary.md

```diff
@@ -1,22 +1,277 @@
 If you have any questions, please email or slack your instructor.
 
 ## Learning Goals
-+ High level understanding of binary
-+ Algorithm to convert from binary to decimal and vice versa
-+ High level overview of memory organization on computers
-  + Bits
-  + Bytes
-  + Computer organization: CPU-memory-buses
-  + Memory address
 
+By the end of this lesson you should be able to:
+
+- Explain the core parts of a computer and what they do
+- Describe the structure of a computer
+- Explain why computers use binary numbers
+- Compare binary numbers to decimal numbers
+- Convert between binary and decimal numbers
+- Explain how high-level languages like Ruby add a level of abstraction to make programming easier.
+
+## Typical Computer Structure
+
+A _Computer_ is a device for storing and processing data according to a series of commands or instructions provided to it.  It is normally structured with a Central Processing Unit, system memory and an I/O bus, which is often called a _motherboard_ connecting them to a variety of peripheral devices (mice, keyboards, etc).  
+
+![Computer System Example](images/computer-system.png)
+
+| Term 	| Definition 	|
+|---	|---	|
+| CPU 	| The **C**entral **P**rocessing **U**nit of a computer.  This is the main processor which executes program instructions, performs arithmetic and coordinates the other components of a computer. 	|
+| Peripherals 	| External components to a computer which can provide input and/or output.  An example is a monitor or keyboard. 	|
+| MAR 	| The Memory Address Register is where the CPU stores an address of an instructor or data in memory.  When data is transferred to or from RAM this is the address used by the CPU.  |
+| MDR | The Memory Data Register is where the CPU stores an information while it is transferred to or from RAM.  |
+| IR 	| The Instruction Register is where the CPU stores the current command to execute.  |
+| PC 	| The Program Counter is where the CPU stores the address of the next instruction to execute.  |
+| Cache 	| A type of very fast memory, close to or on the CPU chip which is used to store information the CPU will likely need to access soon.  It can also be a term for storing information likely to be used in a location which can be quickly accessed.  This technique of pre-staging information likely to be used in proximate locations is called _caching_. 	|
+| Accumulator 	| The accumulator is used to store an input or output operand during execution of the instructions. Operand is the data to be operated by the computer. 	|
+| Register 	| A type of very fast memory, close to or on the CPU chip which is used to store information the CPU will likely need to access soon.  It can also be a term for storing information likely to be used in a location which can be quickly accessed.  This technique of pre-staging information likely to be used in proximate locations is called _caching_. 	|
```

Comment:
This should probably be moved up above the definitions of the specific registers.

Also, this seems to just be a copy/paste of cache.
```suggestion
| Register 	| A type of very small and fast memory on the CPU chip which is used to store information the CPU is currently using. |
```